src/videojs-contrib-hls.js

```diff
@@ -674,13 +674,20 @@ const HlsSourceHandler = function(mode) {
 
       let settings = videojs.mergeOptions(options, {hls: {mode}});
 
+      let previousBeforeRequest;
+
+      if (tech.hls && tech.hls.xhr) {
```

Comment:
In fact, the HLS property on the player is for compatibility reasons. David and I were considering removing the object all together if HLS is used with Video.js 6+